src/tpm_socket_comm.c

```diff
@@ -83,8 +86,9 @@ static void remove_from_buffer(TPM_SOCKET_INFO* socket_info, size_t length)
     }
     else
     {
-        unsigned char* new_buff = (unsigned char*)malloc(socket_info->recv_length - length);
-        memcpy(new_buff, &socket_info->recv_bytes[length], socket_info->recv_length - length);
+        size_t malloc_size = safe_subtract_size_t(socket_info->recv_length, length);
+        unsigned char* new_buff = (unsigned char*)malloc(malloc_size);
```

Comment:
Same here, we should check malloc_size for zero. There was another ticket opened for trying to malloc with zero, which could be result in unpredictable behavior depending on the system.